src/protocol.h

```diff
@@ -412,14 +412,22 @@ class CInv
     std::string ToString() const;
 
     // Single-message helper methods
-    bool IsMsgTx()        const { return type == MSG_TX; }
-    bool IsMsgWtx()       const { return type == MSG_WTX; }
+    bool IsUndefined() const { return type == UNDEFINED; }
+    bool IsMsgTx() const { return type == MSG_TX; }
+    bool IsMsgBlk() const { return type == MSG_BLOCK; }
+    bool IsMsgWtx() const { return type == MSG_WTX; }
+    bool IsMsgFilteredBlk() const { return type == MSG_FILTERED_BLOCK; }
+    bool IsMsgCmpctBlk() const { return type == MSG_CMPCT_BLOCK; }
+    bool IsMsgWitnessBlk() const { return type == MSG_WITNESS_BLOCK; }
     bool IsMsgWitnessTx() const { return type == MSG_WITNESS_TX; }
+    bool IsMsgFilteredWitnessBlk() const { return type == MSG_FILTERED_WITNESS_BLOCK; }
 
     // Combined-message helper methods
-    bool IsGenTxMsg()     const { return type == MSG_TX || type == MSG_WTX || type == MSG_WITNESS_TX; }
+    bool IsGenTxMsg() const { return type == MSG_TX || type == MSG_WTX || type == MSG_WITNESS_TX; }
```

Comment:
Did you consider making these switch statements without defaults, so that the compiler warns if a new type is added to the enum but not here?

Reply:
> Did you consider making these switch statements without defaults, so that the compiler warns if a new type is added to the enum but not here?

That works but requires turning `enum GetDataMsg` into an `enum class` or `enum struct`, which is a fairly noisy change. I'll propose and see what you think.

Reply:
> Did you consider making these switch statements without defaults, so that the compiler warns if a new type is added to the enum but not here?

I think that requires turning `enum GetDataMsg` into an `enum class` or `enum struct`, which is a good but fairly noisy change. I might propose it and see what reviewers think.